### ISSUE TITLE
Implement `verify_data_integrity_proof` for VCs and VPs

### ISSUE THREAD

**Issue by LuisOsta** [knox-networks/ssi#35]

**Comment by LuisOsta** [comment]
This is blocked until we get #29 done

**Comment by fkim7** [comment]
This is from the verifier perspective, to be able to verify proofs from VCs and VPs

**Comment by fkim7** [comment]
@LuisOsta can we think of a way we can demonstrate this process? Would be great to have in our back pocket for technical demos to customers where we show how we can trust these integrity proofs.

**Comment by LuisOsta** [comment]
@fkim7 Yes depending on how you'd want it to be demo'd? I have a couple ideas but did you have anything top of mind?

**Comment by fkim7** [comment]
> @fkim7 Yes depending on how you'd want it to be demo'd? I have a couple ideas but did you have anything top of mind?

Let me know your ideas- I'm not sure but I was thinking variations of printing out different steps/results of the proof generation process, including encryption, hash, etc and then we can hash the body out of band and compare.
